dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/datavalueset/DefaultDataValueSetService.java

```diff
@@ -626,14 +627,31 @@ public ImportSummary importDataValueSetPdf( InputStream in, ImportOptions option
     private ImportSummary importDataValueSet( ImportOptions options, JobConfiguration id,
         Callable<DataValueSetReader> createReader )
     {
+        BatchHandler<DataValue> dvBatch = batchHandlerFactory.createBatchHandler( DataValueBatchHandler.class );
+        BatchHandler<DataValueAudit> dvaBatch = batchHandlerFactory.createBatchHandler(
+            DataValueAuditBatchHandler.class );
+
+        notifier.clear( id );
+
         try ( DataValueSetReader reader = createReader.call() )
         {
-            return importDataValueSet( options, id, reader );
+            ImportSummary summary = importDataValueSet( options, id, reader, dvBatch, dvaBatch );
+
+            dvBatch.flush();
+            dvaBatch.flush();
+
+            NotificationLevel notificationLevel = options.getNotificationLevel( INFO );
+            notifier.notify( id, notificationLevel, "Import done", true )
+                .addJobSummary( id, notificationLevel, summary, ImportSummary.class );
+
+            return summary;
         }
         catch ( Exception ex )
         {
+            dvBatch.flush();
+            dvaBatch.flush();
             log.error( DebugUtils.getStackTrace( ex ) );
-            notifier.clear( id ).notify( id, ERROR, "Process failed: " + ex.getMessage(), true );
```

Comment:
Note that I removed the `clear( id )` part here as I think this wasn't intentionally clearing the messages added so far in case of an error.